### ISSUE TITLE
[list]: Nested Lists contain invalid "treegrid" role structure

### ISSUE THREAD

**Issue by DintaMel** [Esri/calcite-components#14553]
**monday.com sync:** #12170073111

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

List

### Actual Behavior

For nested lists, Axe Devtools is reporting a critical accessibility error because a `role="treegrid"` can't be the child of another `role="treegrid"`. 

Axe Report
<img width="1227" height="637" alt="Image" src="https://github.com/user-attachments/assets/97cb6b4a-2683-4ff3-bff6-461c1e556a54" />

Treegrid role on nested list

<img width="1269" height="274" alt="Image" src="https://github.com/user-attachments/assets/f42fd2f5-869d-472b-a3dd-2a17dfd4847a" />

### Expected Behavior

Each row of a `treegrid` must have the role of `row` or `rowgroup`. See [WAI-ARIA guidance on `treegrid` patterns](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/#:~:text=Each%20row%20container%20has%20role%20row%20and%20is%20either%20a%20DOM%20descendant%20of%20or%20owned%20by%20the%20treegrid%20element%20or%20an%20element%20with%20role%20rowgroup.).

### Reproduction Sample

https://codepen.io/sagewall/pen/xbRYWzP?editors=1000

### Reproduction Steps

1. Open [codepen](https://codepen.io/sagewall/pen/xbRYWzP?editors=1000)
2. Open dev tools
3. Inspect nested List
4. Expand nested list's shadow root. 
5. Expand div's within shadow root. See `role="treegrid" (see screenshot in Actual Behavior section)  

### Reproduction Version

5.0

### Relevant Info

Opened as a result of SF Triage BUG-000185278 where it was initially brought to our attention. On 6/3, the Hub team brought this up in the Get Help - Development Teams channel as well.

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by DintaMel** [comment]
This issue was opened as a result of BUG-000185278 in SF Triage. BUG-000185278 has been accepted. Per guidelines, we should link the issue in SF Triage with this github issue once it is expected to be fixed within the next 2 release cycles.

**Comment by DintaMel** [comment]
+1 from @mjuniper on the hub team

**Comment by geospatialem** [comment]
@DintaMel Added `spike` and `needs more info` for next steps.

Can you update the above issue to reflect the following:
- How did the above surface - through testing, internal discussions, and/or the support ticket, etc.
- Does the above issue dictate next steps for the support ticket or vice versa? If yes, how so? If not, how does it relate?

**Comment by DintaMel** [comment]
> Can you update the above issue to reflect the following:
> 
> * How did the above surface - through testing, internal discussions, and/or the support ticket, etc.

The support ticket was already referenced. I added additional information that the Hub team also surfaced this in Teams. 

> * Does the above issue dictate next steps for the support ticket or vice versa? If yes, how so? If not, how does it relate?

Neither support nor the Hub team dictated how to resolve the axe error, just that they were seeing the Axe error. Can you clarify what additional information you're seeking here?

**Comment by geospatialem** [comment]
> > Can you update the above issue to reflect the following:
> > 
> > * How did the above surface - through testing, internal discussions, and/or the support ticket, etc.
> 
> The support ticket was already referenced. I added additional information that the Hub team also surfaced this in Teams.
> 
> > * Does the above issue dictate next steps for the support ticket or vice versa? If yes, how so? If not, how does it relate?
> 
> Neither support nor the Hub team dictated how to resolve the axe error, just that they were seeing the Axe error. Can you clarify what additional information you're seeking here?

Appreciate the updates. 

During the triage meeting we were:
- Unable to diagnose the severity of the above and how it related to the support ticket since an issue was not linked in from GitHub
- Unable to determine if this issue is dependent or related to the support ticket for next steps. 

Before triaging the above the team needs to determine if this issue is blocking efforts from the support request, vice versa, or work could be performed/investigated independently of one another.

**Comment by DintaMel** [comment]
I took the discussion of the clarification needs to the PE teams chat. There was confusion around this issue being "related to BUG-000185278". As a result, I changed the language to "Opened as a result of SF Triage BUG-000185278". The SF Triage bug was not linked directly to the Github issue because it is not consistent with Esri guidelines to do so.

**Comment by geospatialem** [comment]
A spike needs to be performed to determine for the following:
- What is the current behavior with assistive technologies, is it behaving as expected?
- The potential resolution and what is expected for assistive technologies, in addition to the current impact on assistive technologies
